src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -510,9 +510,9 @@ void CoreWorkerDirectTaskReceiver::HandleTask(
             return_object->set_metadata(result->GetMetadata()->Data(),
                                         result->GetMetadata()->Size());
           }
-          for (const auto &nested_id : result->GetNestedIds()) {
-            return_object->add_nested_inlined_ids(nested_id.Binary());
-          }
+        }
+        for (const auto &nested_id : result->GetNestedIds()) {
```

Comment:
I am confused about this change. It means we can have inlined nested IDs where the outer id is a plasma object?

Reply:
yup, I assume that's necessary for both inlined and plasma objects.

Reply:
Are we testing both cases? It looks like the test case is only for outer_id : inlined? (lmk if I misunderstand this)

Reply:
Oops I guess it is merged before it is handled?

Reply:
Ah my bad. I think it's fine without a test, but I'll leave it to @scv119 to decide.